File: compy_cli/test_dir/python/src/classes/nested_dependencies.py
class _ClassA:
    def __init__(self, a: int) -> None:
        self.a = a


class _ClassB:
    def __init__(self, b: int, class_a: _ClassA) -> None:
        self.b = b
        self.class_a = class_a


class _ClassC:
    def __init__(self, c: int, class_b: _ClassB) -> None:
        self.c = c
        self.class_b = class_b

    def access(self) -> int:
        return self.class_b.class_a.a


def class_nested_dependencies_fn():
    print("CLASS NESTED DEPENDENCIES RESULTS:")
    class_a: _ClassA = _ClassA(1)
    class_b: _ClassB = _ClassB(2, class_a)
    class_c: _ClassC = _ClassC(3, class_b)
    print(class_c.access())

File: pypp_cli/test_dir/cpp/src/classes/nested_dependencies.cpp
#include "classes/nested_dependencies.h"
#include "py_str.h"
#include "pypp_util/print.h"
#include <utility>

class _ClassA {
  public:
    int a;
    _ClassA(int a_a) : a(std::move(a_a)) {}
};

class _ClassB {
  public:
    int b;
    _ClassA &class_a;
    _ClassB(int a_b, _ClassA &a_class_a)
        : b(std::move(a_b)), class_a(a_class_a) {}
};

class _ClassC {
  public:
    int c;
    _ClassB &class_b;
    _ClassC(int a_c, _ClassB &a_class_b)
        : c(std::move(a_c)), class_b(a_class_b) {}
    int access() { return class_b.class_a.a; }
};

void class_nested_dependencies_fn() {
    pypp::print(pypp::PyStr("CLASS NESTED DEPENDENCIES RESULTS:"));
    _ClassA class_a = _ClassA(1);
    _ClassB class_b = _ClassB(2, class_a);
    _ClassC class_c = _ClassC(3, class_b);
    pypp::print(class_c.access());
}
